whylog/teacher/__init__.py

```diff
@@ -1,49 +1,78 @@
 from whylog.teacher.mock_outputs import create_sample_rule
-from whylog.teacher.user_intent import UserRuleIntent
+
+
+class TeacherParser(object):
+    def __init__(self, line_object):
+        self.line = line_object
+        self.pattern_name = None
+        self.log_type = None
+        self.primary_keys = []
+
+
+class TeacherConstraint(object):
+    def __init__(self):
+        self.constr_type = None
```

Comment:
Is constraint type unknown during the process of creation of this object, so that we need to initialize it as `None` and set later? Maybe it could just be stored since object creation?

Please try to not abbreviate every variable name.